world/petitions/petitions_commands.py

```diff
@@ -7,9 +7,204 @@
+        matched_list = []
+        for ad in ads:
+            matches = ad.match(self.caller.dompc)
+            if matches and (ad.TIME in matches):
+                matches_list.append(matches)
+                matched_list.append(ad)
+        table = evtable.EvTable(
+            "Name/Gender/Age",
+            "Matches",
+            "Times",
+            "Blurb",
+            border="cells",
+            width=78)
+        if len(matches_list) > 0 and len(matched_list):
+            for match, ad in zip(matches_list, matched_list):
+                matchstring = ""
+                for m in match:
+                    matchstring += "[{}]".format(ad.MATCH_TYPES[m][1])
+                table.add_row("{}/{}/{}".format(ad.owner.player.char_ob.key,
+                                                ad.owner.player.char_ob.db.gender[0],
+                                                ad.owner.player.char_ob.db.age),
+                              matchstring,
+                              "{}-{}".format(ad.playtime_start,
+                                             ad.playtime_end),
+                              ad.blurb)
+        table.reformat_column(2, width=8)
+        table.reformat_column(1, width=18)
+        table.reformat_column(0, width=20)
+        self.msg(table)
+
+    def display_all_matches(self):
+        my_ad = self.caller.dompc.wanted_ad
+        ads = WantedAd.objects.filter(active=True)
+        table = evtable.EvTable(
+            "Name/Gender/Age",
+            "Matches",
+            "Times",
+            "Blurb",
+            border="cells",
+            width=78)
+        for ad in ads:
+            match = ad.match(self.caller.dompc)
+            matchstring = ""
+            for m in match:
+                matchstring += "[{}]".format(ad.MATCH_TYPES[m][1])
+            table.add_row("{}/{}/{}".format(ad.owner.player.char_ob.key,
+                                            ad.owner.player.char_ob.db.gender[0],
+                                            ad.owner.player.char_ob.db.age),
+                          matchstring,
+                          "{}-{}".format(ad.playtime_start,
+                                         ad.playtime_end),
+                          ad.blurb)
+        table.reformat_column(2, width=8)
+        table.reformat_column(1, width=18)
+        table.reformat_column(0, width=20)
+        self.msg(table)
+
+    def clues(self):
+        ad = self.caller.dompc.wanted_ad
+        clues = []
+        args = self.args.split("/")
+        message = "You're now matching: "
+        if not args:
+            ad.clues.clear()
+            message = ("You're now matching no clues")
+        else:
+            ad.clues.clear()
+            for entry in args:
+                try:
+                    clue = self.get_by_name_or_id(Clue, entry)
+                    ad.clues.add(clue)
+                    for tag in clue.search_tags.all():
+                        ad.searchtags.add(tag)
```

Comment:
Another situation of db operations in loops, but this probably wouldn't be as problematic. Still, would be better to get all the tags collectively with like `SearchTag.objects.filter(id__in=[ob.id for ob in clues])` if you got all the clues beforehand, for example, then add them to the ad.

Reply:
Does it make a difference? Wouldn't the filter have to do all the same operations that this loop is doing? Either way I'll change it.